generators/client/index.js

```diff
@@ -315,6 +315,7 @@ module.exports = JhipsterClientGenerator.extend({
         writeSwaggerFiles: function () {
             // Swagger-ui for Jhipster
             this.template(MAIN_SRC_DIR + 'swagger-ui/_index.html', MAIN_SRC_DIR + 'swagger-ui/index.html', this, {});
+            this.template(MAIN_SRC_DIR + 'swagger-ui/_index-prod.html', MAIN_SRC_DIR + 'swagger-ui/index-prod.html', this, {});
```

Comment:
may be you could use the `inject` plugin to do this dynamically based on profile.